docs/wiki/CL/overview.md

```diff
@@ -1,20 +1,415 @@
+
+However, both PoW and PoS are closely linked to the consensus mechanisms they support through [fork choice](/wiki/link-to-fork-choice-cl-arch) rules. They help assign a weight or score to a chain of blocks: in PoW, it's the total computational work done; in PoS, it's the total value staked that supports a particular chain. Beyond these basics, PoW and PoS can support various consensus protocols, each with its own dynamics and trade-offs.
+
+### Block chains
+
+The basic element of a blockchain is the block. A block contains a set of transactions assembled by a leader (block proposer). The contents of a block can vary based on the protocol.
+
+- The payload of a block on Ethereum's execution chain is a list of user transactions.
+- In the pre-Merge PoS Beacon Chain, a block's payload was mostly a set of attestations by validators.
+- Post-Merge Beacon Chain blocks also include the execution payload (user transactions).
+- After [EIP-4844](https://eips.ethereum.org/EIPS/eip-4844)(Deneb upgrade), blocks now also contain commitments to opaque data blobs alongside user transactions.
+
+Except for the Genesis block, each block builds on and points to a parent block, forming a chain of blocks: a blockchain. The goal is for all nodes on the network to agree on the same blockchain history.
+
+<a id="img_blockchain"></a>
+
+<figure class="diagram" style="text-align:center">
+
+![Diagram for Block Chain](../../images/cl/blockchain.svg)
+
+<figcaption>
+
+_Time moves from left to right and, except for the Genesis block, each block points to the parent block it builds on._
+
+</figcaption>
+</figure>
+
+The chain grows as nodes add new blocks to its tip. This is done by temporarily selecting a "leader", the node that extends the chain. In PoW, the leader is the miner who first solves the PoW puzzle for its block. In Ethereum's PoS, the leader is pseudo-randomly selected from active stakers.
+
+The leader (block proposer) adds a block to the chain, choosing and ordering its contents. The block must be valid according to protocol rules, or the network will ignore it. Using blocks is an optimization. Adding individual transactions one by one would create a huge consensus overhead. So blocks are batches of transactions, and sometimes [people argue](https://www.bitrawr.com/bitcoin-block-size-debate-explained) about how big those blocks should be. The size of these blocks can vary:
+
+In Bitcoin, block size is limited by the number of bytes. In Ethereum's execution chain, block size is limited by the block gas limit (the amount of work needed to process the transactions). [Beacon block](link to beaconblock class) sizes are limited by hard-coded constants.
+
+### Forks and ChainId
```

Comment:
I would omit this section. Chaind is a basic network spec which is independent of consensus mechanism and I don't see how it's relevant here

Reply:
Omitted this